src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -291,6 +291,11 @@ public static bool IsNumericType([NotNullWhen(returnValue: true)] this ITypeSymb
                     case SpecialType.System_Single:
                     case SpecialType.System_Double:
                     case SpecialType.System_Decimal:
+#if !CODE_STYLE // TODO: Remove the #if once IsNativeIntegerType is available.
+                // https://github.com/dotnet/roslyn/issues/41462 tracks adding this support
+                    case SpecialType.System_IntPtr when type.IsNativeIntegerType:
+                    case SpecialType.System_UIntPtr when type.IsNativeIntegerType:
```

Comment:
:memo: I'm not sure how to cover those cases, but made the change opportunistically anyway

Reply:
interesting that it's not a special SpecialType.  i.e. why not SpecialType.Nint?

Reply:
I'll let @cston answer for design of nint type symbol.

Reply:
`SpecialType` has a very specific definition, and it did not seem appropriate to change that. See [https://github.com/dotnet/roslyn/blob/master/src/Compilers/Core/Portable/SpecialType.cs#L17-L24](https://github.com/dotnet/roslyn/blob/master/src/Compilers/Core/Portable/SpecialType.cs#L17-L24)

---
In reply to: [401951786](https://github.com/dotnet/roslyn/pull/42999#discussion_r401951786) [](ancestors = 401951786)

Reply:
@cston As a consumer of the API, how does that specific definition make sense? This feels like an implementation detail leaking in some way?